build/build.cake

```diff
@@ -0,0 +1,237 @@
+    });
+
+    Information("\nChecking " + files.Count() + " file header(s)");
+    foreach(var file in files)
+    {
+        var oldContent = FileReadText(file);
+        var rgx = new Regex("^(//.*\r?\n|\r?\n)*");
+        var newContent = header + rgx.Replace(oldContent, "");
+
+        if(!newContent.Equals(oldContent, StringComparison.Ordinal))
+        {
+            if(Replace)
+            {
+                Information("\nUpdating " + file + " header...");
+                FileWriteText(file, newContent);
+            }
+            else
+            {
+                Error("\nWrong/missing header on " + file);
+                hasMissing = true;
+            }
+        }
+    }
+
+    if(!Replace && hasMissing)
+    {
+        throw new Exception("Please run '.\\build.ps1 -target=UpdateHeaders' and commit the changes.");
+    }
+}
+
+//////////////////////////////////////////////////////////////////////
+// Default Task
+//////////////////////////////////////////////////////////////////////
+
+Task("Clean")
+    .Description("Clean the output folder")
+    .Does(() =>
+{
+    if(DirectoryExists(binDir))
+    {
+        Information("\nCleaning Working Directory");
+        CleanDirectory(binDir);
+    }
+    else
+    {
+        CreateDirectory(binDir);
+    }
+});
+
+Task("Verify")
+    .Description("Run pre-build verifications")
+    .IsDependentOn("Clean")
+    .Does(() =>
+{
+    VerifyHeaders(false);
+});
+
+Task("Version")
+    .Description("Updates the version information in all Projects")
+    .IsDependentOn("Verify")
+    .Does(() =>
+{
+    Information("\nDownloading NerdBank GitVersioning...");
+    var installSettings = new NuGetInstallSettings {
+        ExcludeVersion  = true,
+        Prerelease = true,
+        Version = "2.0.37-beta",
+        OutputDirectory = toolsDir
+    };
+    
+    NuGetInstall(new []{"nerdbank.gitversioning"}, installSettings);
+
+    Information("\nRetrieving version...");
+    var results = StartPowershellFile(versionClient);
+    Version = results[1].Properties["NuGetPackageVersion"].Value.ToString();
+    Information("\nBuild Version: " + Version);
+});
+
+Task("Build")
+    .Description("Build all projects and get the assemblies")
+    .IsDependentOn("Version")
+    .Does(() =>
+{
+    Information("\nBuilding Solution");
+    var buildSettings = new MSBuildSettings
+    {
+        MaxCpuCount = 0
+    }
+    .SetConfiguration("Release")
+    .WithTarget("Restore");
+
+    // Force a restore again to get proper version numbers https://github.com/NuGet/Home/issues/4337
+    MSBuild(Solution, buildSettings);
+    MSBuild(Solution, buildSettings);
+
+    buildSettings = new MSBuildSettings
+    {
+        MaxCpuCount = 0
+    }
+    .WithTarget("Build")
+    .WithProperty("GenerateSolutionSpecificOutputFolder", "true")   
+    .WithProperty("GenerateLibraryLayout", "true")
+    .WithProperty("TreatWarningsAsErrors", "false")
+    .WithProperty("PackageOutputPath", nupkgDir);
```

Comment:
@onovotny This is the line where I was attempting to Generate Packages, I will double check it later.

Reply:
yes. It's also critical that it restore twice, separately, before the build step.

Reply:
All the properties/steps are here:
https://github.com/Microsoft/UWPCommunityToolkit/blob/dev/build/default.ps1#L117-L120

Reply:
@onovotny Why is Package Output Path in the PS Script in Quote marks? I can't seem to get it to generate Packages to that folder.